### ISSUE TITLE
import losing search params

### ISSUE THREAD

**Issue by markcarroll** [remcohaszing/rehype-mdx-import-media#1]
I am trying to use this in NextJS app router with mdx files. I have the img tags being replaced by a custom component that reads the search param from the path and uses it to size the image. However, when I add this plugin, while the local image load works nicely, I lose the query param.  Can it be preserved in the path, or moved to a different attribute?

Example MDX:

```md
![alt text](images/IMG_1641-2.jpg?w=full)
![alt text](images/IMG_1643.jpg?w=400&h=300)
```

If there is a way to add more attributes, that is not clear from the documentation. Any help appreciated.

**Comment by remcohaszing** [comment]
If you share code, use a markdown code block, not an image. That’s better for accessibility and just general convenience.

---

Anyway, your idea is clear and it makes sense. Currently query parameters are available on the import source, not on the JSX prop. It’s useful if you can choose to preserve it on either, both, or none. I propose a new option:

- `preserveQuery` (`'both' | 'import' | 'jsx' | 'none'`): Whether to keep query parameters on the import declaration or on the JSX prop. (Default `'import'`).

How should the URL hash behave? Should it be at the same location as the query parameters? Should it be a separate option?

---

All logic needed to support this is in the `getIdentifier` function. Would you like to provide a PR?

---

A separate rehype plugin could add additional attributes. `rehype-mdx-import-media` will preserve them.

**Comment by markcarroll** [comment]
Sorry about the markdown issue. I completely missed that.  - fixed

Yes I like the `preserveQuery` idea, that would work well.  In my case I would prefer that the attributes get preserved on the jsx, so I'd be setting to `jsx`. There are really no other ways of getting preferences from markdown images to a component, post MDX processing. 

As a work around I am using the title field to pass the props through but that is obviously not a good solution.

**Comment by remcohaszing** [comment]
Released in [v1.2.0](https://github.com/remcohaszing/rehype-mdx-import-media/releases/tag/v1.2.0)! :tada:
